lua/lspconfig/server_configurations/remark_ls.lua

```diff
@@ -11,7 +11,16 @@ return {
   default_config = {
     cmd = cmd,
     filetypes = { 'markdown' },
-    root_dir = util.find_git_ancestor,
+    root_dir = util.root_pattern(
+      '.remarkc',
+      '.remarkc.json',
+      '.remarkc.cjs',
+      '.remarkc.mjs',
+      '.remarkc.js',
+      '.remarkc.yaml',
+      '.remarkc.yml',
+      'package.json'
```

Comment:
This should really be `package.json` AND which has a field `remarkConfig`. I don't believe there's a util function for this currently?

I wonder if it'd make sense to make `root_pattern` accept a table where the first index is the filename, and then the rest are keys with extra conditions, like:

```suggestion
      { 'package.json', withKey = { "remarkConfig" } }
```

or (if this were to even be added) it'd deserve its own util function

Reply:
This would benefit other tools as well (at least ESLint and Stylelint).